src/components/ArticlesPage.jsx

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import { getArticles } from "../api";
+import Button from 'react-bootstrap/Button';
+import Card from 'react-bootstrap/Card';
+import { Link } from "react-router-dom";
+import { Spinner } from "react-bootstrap";
+
+export const ArticlesPage=() => {
+    const [articles, setArticles] = useState([]);
+    const [loading, setLoading] = useState(true);
+  
+    
+  useEffect(() => {
+    getArticles()
+      .then(({articles}) => {
+        setArticles(articles);
+        setLoading(false)
+      })
+      .catch((err) => {console.log(err)});
+  }, []);
+
+  
+  if (loading) {
+    return (
+      <div className="loading-container">
+        <Spinner animation="border" role="status">
+          <span className="visually-hidden">Loading...</span>
+        </Spinner>
+      </div>
+    );
+  }
+  
```

Comment:
Nice, but if this is the loader you wish to use in other places as well I would recommend creating a component for it